### PR TITLE
Fix logging version; support .yaml extension

### PR DESCRIPTION
**Pull request by snukky**
Changes:
* Fixing the message "Model created with Marian false"
* Logging Marian version for newly started training
* Adding .yaml as a supported vocab extension

### CODE REVIEW

File: src/common/config_parser.cpp [hunk 1]
```diff
@@ -44,8 +44,7 @@ void ConfigParser::addOptionsGeneral(cli::CLIWrapper& cli) {
 
   // clang-format off
   cli.add<bool>("--version",
-     "Print version number and exit",
-     false);
+     "Print version number and exit");
```
Comment (emjotde) [inline src/common/config_parser.cpp]: Is this false by default?

File: src/common/config_parser.cpp [hunk 2]
```diff
@@ -635,6 +634,8 @@ void ConfigParser::parseOptions(int argc, char** argv, bool doValidate) {
   if(get<bool>("version")) {
     std::cerr << PROJECT_VERSION_FULL << std::endl;
     exit(0);
+  } else {
+    config_["version"] = PROJECT_VERSION_FULL;
```
Comment (emjotde) [inline src/common/config_parser.cpp]: Maybe add a comment. It took me a moment to understand what this does.

File: src/common/config_parser.cpp (hunk 1 above)
Comment (snukky) [reply src/common/config_parser.cpp]: I removed the definition of --version from ConfigParser. Now it's automatically added like --help is and not present in `config`

File: src/common/config_parser.cpp (hunk 2 above)
Comment (snukky) [reply src/common/config_parser.cpp]: Now, `config["version"]` is a version number and not related to the --version flag

File: src/common/config.cpp [hunk 3]
```diff
@@ -63,10 +58,31 @@ void Config::initialize(int argc, char** argv, cli::mode mode, bool validate) {
 
   log();
 
-  if(has("version"))
+  // Log version of Marian that has been used to create the model.
+  //
+  // Key "version" is present only if loaded from model parameters and is not
+  // related to --version flag
+  if(has("version")) {
+    auto version = get("version").as<std::string>();
```
Comment (emjotde) [inline src/common/config.cpp]: Better use `get<std::string>("version")`

File: src/common/config.cpp [hunk 4]
```diff
@@ -63,10 +58,31 @@ void Config::initialize(int argc, char** argv, cli::mode mode, bool validate) {
...
+  }
+  // If this is a newly started training
+  else if(mode == cli::mode::training) {
     LOG(info,
-        "[config] Model created with Marian {}",
-        get("version").as<std::string>());
+        "[config] The model is being created with Marian {}",
+        PROJECT_VERSION_FULL);
+    config_["version"] = PROJECT_VERSION_FULL;
+  }
```
Comment (emjotde) [inline src/common/config.cpp]: I think we should log the Marian version every time, not only during training?

File: src/common/config.cpp (hunk 3 above)
Comment (snukky) [reply src/common/config.cpp]: Fixed

File: src/common/config.cpp (hunk 4 above)
Comment (snukky) [reply src/common/config.cpp]: It's also logged for translation with a Marian model. We have a regression test for this.

I see these cases: 
* (A) If a loaded model has been trained with Marian, the version will be loaded from the model configuration file and logged for training/translation/scoring.
* If there is no model configuration file, it means that it's either (B) a newly started training (message "Model is being created with...")
* or (C) translation/scoring with a model trained with Nematus or a very old Marian (no message).

"[config] version: xyz" will be displayed only if a version is loaded from the model configuration file.

**Pull request merged**